src/components/ContrastContent.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { ReactNode } from 'react';
+import { createMuiTheme, MuiThemeProvider } from '@material-ui/core';
+import theme from '../themes/mainTheme';
+
+interface ContrastContentProps {
+  backgroundColor: string;
+  children: ReactNode;
+}
+
+const ContrastContent = ({ backgroundColor, children }: ContrastContentProps) => {
+  const contrastTextColor = theme.palette.getContrastText(backgroundColor);
+
+  const contrastTheme = createMuiTheme({
+    ...theme,
+    palette: {
+      ...theme.palette,
+      primary: { main: contrastTextColor },
+    },
+  });
+
+  return <MuiThemeProvider theme={contrastTheme}>{children}</MuiThemeProvider>;
```

Comment:
children her kan feks være `Typography`? Her settes palette.primary i theme, men Typography bruker sikkert `color="textPrimary"` som default(?)
uansett: hvordan kan vi håndtere at color kan bli satt til `textPrimary`, `textSecondary`, `primary` og hele røkla? alle de burde jo være ulik, i tillegg til at de må passe til bakgrunnsfargen

begynner å lure på om det er så lurt å i heletatt tillate bakgrunnsfarger som ikke funker uten å trikse med kontrastfarger... 🤔

Reply:
ja, `Typography, Checkbox, RadioButtons` osv. Alle de blir påvirket av at primary settes til `contrastTextColor`. 
Hvis vi skriver `<Typography color="primary">Contrast</Typography>`, så vil `Contrast`-teksten settes til `contrastTextColor` hvis den `Typography`-noden er en child av `BackgroundDiv`.

skjønner ikke helt uansett-spørsmålet ditt. Vi kontrollerer jo kun children av `BackgroundDiv`.

Reply:
Spørsmålet er egentlig for alle mulige komponenter:
`<Button color="primary">`
`<Button color="secondary">`
`<Typography color="textSecondary">`
osv

hvordan kan vi vite at alle ulike verdier for color passer for background når vi bare setter primary? Tenker da også på alt som er "skjult" for oss, ting som skjer på hover, fokus, etc. Det eneste vi vet her er at `primary` vil ha en lovlig verdi gitt background.

eeegentlig er det kanskje 2 themes vi trenger `light` og `dark`, så vi kan kanskje bruke `palette.type` til noe her også?

Reply:
![Skjermbilde 2021-01-21 kl  11 03 34](https://user-images.githubusercontent.com/6595982/105335457-59a54900-5bd8-11eb-8396-8c4dff6e9c0e.png)

Sånn ser det ut hvis denne koden er en child av `BackgroundDiv`:
```
<Button color="primary" variant="outlined">
        knappetekst
</Button>
```

Reply:
Husk at `BackgroundDiv` kun brukes i Wizarden nå da, og det er jo selvfølgelig mulig å endre på den. Men for denne oppgaven og denne PRen, så synes jeg den oppfyller kravene som er satt. Ja, vi skal muligens ha dark theme i fremtiden, men jeg synes ikke dette er noe som er relevant for denne PRen

Reply:
Dessuten så skal jeg starte å jobbe med tekster nå, så `BackgroundDiv` blir mest sannsynlig endret snart uansett (til å bruke flere enn kun "primary"-fargen).

Reply:
men hva hvis `color="secondary"`? secondary endres aldri så hvordan kan vi vite at secondary, default, textPrimary, textSecondary, osv også virker med alle bakrunnene?

Ang dark/light theme så tenkte jeg på bakrunnene: det finnes kanskje en grense der, feks hvis bakgrunnen er dark eller megaDark så kan vi bare sette `type: "dark"` i theme, så fikser MUI mye av resten selv... men nå skyter jeg bare fra hofta 😅

Reply:
Ja, jeg er helt enig i at noe mer må gjøres. Og det kommer deg nok til å gjøres i neste PR fra meg. Men ikke noen vits i å konvertere alle fargene vi _kanskje_ trenger i fremtiden. Er det ikke bedre å gjøre litt og litt? Så når vi legger til en knapp i wizarden som har secondary-farge, så ser vi jo fort at BackgroundDiv må endres på? Det er jo ikke en fasit sånn som det er nå. Alt endres med tiden.

Reply:
Er bare redd alle cornercasene som vil dukke opp, for disse bakgrunnene skal jo inn på landing page, min profil og overalt ellers også sikkert. worst case er jo at vi ikke får til en god løsning som funker overalt, så må Kine skissere på nytt og vi må implementere på nytt. Men vi kan prøve litt fremover så ser vi 👍

Reply:
Jeg må ærlig talt si at jeg ikke helt skjønner bekymringen din altså. Hvis vi finner ut om 2 mnd at dette ikke fungerer lengre, så bytter vi det jo ut med noe annet? Blir nok enklere når Typography tillater custom colors i color-propen også.